bridges/otelzap/encoder.go

```diff
@@ -93,11 +95,24 @@ func (m *objectEncoder) AddUint64(k string, v uint64) {
 		})
 }
 
-// TODO.
+// AddReflected converts all non-primitive type to JSON string.
 func (m *objectEncoder) AddReflected(k string, v interface{}) error {
+	// TODO use json Encoder from a pool.
+	enc := json.NewEncoder(m)
+	if err := enc.Encode(v); err != nil {
+		return err
+	}
+	m.AddString(k, m.reflectval.AsString())
```

Comment:
Shouldn't we do something like here: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/b9072109ea81db7e384917c9dd4528623e44e09b/bridges/otellogrus/hook.go#L186-L237?

Reply:
actually `zap` does this conversion internally. It falls back to reflect only if necessary. Doing this mapping again would be overkill right? 

https://github.com/uber-go/zap/blob/273f9ea8c8870d5f468f33f5026cd2e7ed4bd061/field.go#L473

Reply:
Nice.

Still, wouldn't this part would be still helpful?

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/b9072109ea81db7e384917c9dd4528623e44e09b/bridges/otellogrus/hook.go#L202-L231